### PR TITLE
Fixes #13571 : Introduce webapp tests to mattermost-plugin-starter-template

### PR DESCRIPTION
**Pull request by M-ZubairAhmed**
#### Summary
This Pull requests adds the necessary boilerplate for including the tests in template repository.

#### Ticket Link
Fixes : https://github.com/mattermost/mattermost-server/issues/13571
Jira : https://mattermost.atlassian.net/browse/MM-18539

### CODE REVIEW

**Comment by M-ZubairAhmed** [comment]
I have added a sample React component and a test, to see if the jest is working or not and its works

File: webapp/package.json [hunk 1]
```diff
@@ -23,32 +24,86 @@
     "@babel/preset-react": "7.0.0",
     "@babel/preset-typescript": "7.3.3",
     "@babel/runtime": "7.4.5",
+    "@types/enzyme": "^3.10.4",
```
Comment (mickmister) [inline webapp/package.json]: Can we use exact versions by removing the `^` characters on these dependency versions, including existing dependencies?
```suggestion
    "@types/enzyme": "3.10.4",
```

Comment (M-ZubairAhmed) [reply webapp/package.json]: Got it, will fix this

Comment (mickmister) [reply webapp/package.json]: Do you mind applying this change to all dependencies in `package.json`, then run `npm install`?

File: webapp/package.json [hunk 2]
```diff
@@ -23,32 +24,86 @@
...
-    "@types/enzyme": "3.10.3",
-    "@types/jest": "24.0.17",
-    "@types/node": "12.7.0",
-    "@types/react": "16.8.24",
-    "@types/react-dom": "16.8.5",
-    "@types/react-router-dom": "4.3.4",
-    "@types/react-transition-group": "4.2.2",
-    "@typescript-eslint/parser": "1.13.0",
+    "core-js": "^3.6.2",
+    "mattermost-webapp": "github:mattermost/mattermost-webapp#23f5f93d9f12a7e2b5623e5cee6814366abd9a0f",
```
Comment (marianunez) [inline webapp/package.json]: Do we want all plugins to have a dependency on the webApp? Is this required?

Comment (mickmister) [reply webapp/package.json]: It's a devDependency, used only for a one-liner to setup the tests. It's not imported in the plugin's source anywhere.

`import 'mattermost-webapp/tests/setup';`

https://github.com/M-ZubairAhmed/mattermost-plugin-starter-template/blob/4eda4517d417ad9658b7c527109cca698b022cea/webapp/tests/setup.js#L4

**Comment by mickmister** [comment]
Confirmed that tests run properly :+1:

**Pull request merged**